shell/platform/linux/fl_platform_plugin.cc

```diff
@@ -234,15 +237,32 @@ static void request_app_exit_response_cb(GObject* object,
   }
 }
 
-// Send a request to Flutter to exit the application.
+// Send a request to Flutter to exit the application, but only if it's ready for
+// a request.
 static void request_app_exit(FlPlatformPlugin* self, const char* type) {
   g_autoptr(FlValue) args = fl_value_new_map();
+  if (!self->app_initialization_complete ||
+      g_str_equal(type, kExitTypeRequired)) {
+    quit_application();
+    return;
+  }
+
   fl_value_set_string_take(args, kExitTypeKey, fl_value_new_string(type));
   fl_method_channel_invoke_method(self->channel, kRequestAppExitMethod, args,
                                   self->cancellable,
                                   request_app_exit_response_cb, self);
 }
 
+// Called when the Dart app has finished initialization and is ready to handle
+// requests. For the Flutter framework, this means after the ServicesBinding has
+// been initialized and it sends a System.initializationComplete message.
+static FlMethodResponse* system_intitialization_complete(
+    FlPlatformPlugin* self,
+    FlMethodCall* method_call) {
+  self->app_initialization_complete = TRUE;
+  return nullptr;
```

Comment:
Shouldn't this return `FL_METHOD_RESPONSE(fl_method_success_response_new(nullptr))` - otherwise this method call never returns.

Reply:
Yes, good point.  I was thinking that it was basically a "void" call, but it should respond.